tools/benchmark/src/ResultTypes.ts

```diff
@@ -0,0 +1,134 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type * as v8 from "node:v8";
```

Comment:
This import and the types that depend on it need to move back to `memoryTestRunner.ts` (only the ones that really depend on this, i.e. `MemoryTestData` and `MemorySampleData`). It was deliberate that this dependency on node is only in the memory test runner, and we know that means memory tests aren't supported when using the benchmark tool in the browser. But this file has types related to runtime benchmarks, so it will be loaded when someone tries to use the tool in a browser, and would fail if we keep this here.